main.tf

```diff
@@ -1,13 +1,50 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled = module.this.enabled
+
+  // manage quotas that have a `value` defined
+  quotas_to_manage         = { for i, quota in var.service_quotas : i => quota if lookup(quota, "value", null) != null }
+  quotas_to_manage_by_code = { for i, quota in local.quotas_to_manage : i => quota if lookup(quota, "quota_code", null) != null }
+  quotas_to_manage_by_name = { for i, quota in local.quotas_to_manage : i => quota if lookup(quota, "quota_name", null) != null }
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
-  }
+  // lookup quotas that have a `quota_code` or `quota_name` defined and `value` is null
+  quotas_to_lookup         = { for i, quota in var.service_quotas : i => quota if lookup(quota, "value", null) == null }
+  quotas_to_lookup_by_code = { for i, quota in local.quotas_to_lookup : i => quota if lookup(quota, "quota_code", null) != null }
+  quotas_to_lookup_by_name = { for i, quota in local.quotas_to_lookup : i => quota if lookup(quota, "quota_name", null) != null }
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+# for_each service_quota where `value` is not `null`, try to modify
+resource "aws_servicequotas_service_quota" "managed_by_code" {
+  for_each     = local.enabled ? local.quotas_to_manage_by_code : {}
```

Comment:
did you try this with `enabled = false`?
terraform complains in many cases that both sides of the `?` operator must have the same type and the same types of all items in the map.

To get around this, we usually do 

```
quotas_to_manage   = { for i, quota in var.service_quotas : i => quota if local.enabled && lookup(quota, "value", null) != null }

```

and then

```
for_each     = local.quotas_to_manage_by_code
```